src/coord/src/coord.rs

```diff
@@ -2182,17 +2210,51 @@ impl Coordinator {
                                 })
                                 .collect();
 
+                            if let CatalogItem::Table(t) = catalog_entry.item_mut() {
+                                if let Some(persist) = &mut t.persist {
+                                    let updates: Vec<((Vec<u8>, ()), u64, isize)> = updates
+                                        .iter()
+                                        .map(|u| {
+                                            let mut encoded_row = Vec::new();
+                                            u.row.encode(&mut encoded_row);
+                                            ((encoded_row, ()), u.timestamp, u.diff)
+                                        })
+                                        .collect();
+                                    let tx = tx.clone();
+                                    let callback =
+                                        Box::new(move |res: Result<SeqNo, PersistError>| {
+                                            let res =
+                                                res.map(|_| ()).map_err(|err| err.to_string());
+                                            if let Err(err) = tx.send(res) {
+                                                log::debug!(
+                                                    "error notifying insert of completion: {}",
+                                                    err
+                                                );
+                                            }
+                                        });
+                                    persist
+                                        .write_handle
+                                        .write(&updates, CmdResponse::Callback(callback));
+                                }
+                            }
+                            // TODO: Move this to happen in the callback when we're persisting.
```

Comment:
the "write -> tail with no SequencedCommand::Insert" architecture actually solves this TODO pretty nicely. i might take a shot at making that work today and getting it in before this PR